packages/amplify-cli/src/extensions/amplify-helpers/remove-env-from-cloud.ts

```diff
@@ -31,12 +31,12 @@ export const removeEnvFromCloud = async (context, envName, deleteS3): Promise<vo
   try {
     await Promise.all(providerPromises);
     await raiseInternalOnlyPostEnvRemoveEvent(context, envName);
-  } catch (e) {
-    if (e.code !== 'NotFoundException') {
+  } catch (ex) {
+    if (!(ex?.name === 'BucketNotFoundError')) {
```

Comment:
just a nit, it might not be the case but if the `ex` is not instance of `AmplifyException`, it will throw an error as it will try to access a property that doesn't exist, if we keep this version can we use `(ex?.name !== 'BucketNotFoundError')` to remove some braces